public/js/build/BuildRecipeViewController.js

```diff
@@ -52,14 +53,52 @@ class BuildRecipeViewController {
         //Add an "Info" button
         let infoButton = document.createElement('button');
         infoButton.textContent = "Info";
+        infoButton.setAttribute("id", "info-" + ingredient_id);
+
         // When user clicks on Info button, ingredient info show ups asynchronously on page
         infoButton.addEventListener('click', function() {
           function myCallback(result) {
             console.log(result);
             // Update Ingredient Info box on page async
             document.getElementById("ingredient-name").textContent = result.ingredient;
             document.getElementById("ingredient-ethics").textContent = result.problem;
-            document.getElementById("ingredient-alternatives").innerHTML = result.alternative.join("<br>");
+
+            // Populate alternatives table with names and buttons
+            let altTable = document.getElementById("alt-table");
+            altTable.innerHTML = "";
+            if (result.alternative != "None") {
+              for (i = 0; i < result.alternative.length; i++) {
+                let altRow = altTable.insertRow();
+
+                // Add a cell to hold alternative name
+                let altNameCell = altRow.insertCell();
+                altNameCell.appendChild(document.createTextNode(result.alternative[i]));
+
+                // Add a cell to hold 'Add' button
+                let altAddCell = altRow.insertCell();
+                let altAddButton = document.createElement("button");
+                let altAddFunction = ff.createAddIngredientFunction(result.alternative_id[i], result.alternative[i], self);
+                altAddButton.textContent = "Add";
+                altAddButton.addEventListener("click", altAddFunction);
+                altAddCell.appendChild(altAddButton);
+
+                // Add a cell to hold 'Replace' button
+                let altReplaceCell = altRow.insertCell();
+                let altReplaceButton = document.createElement("button");
+                let altReplaceFunction = function() {
+                  let removeButton = document.getElementById("remove-" + result.ingredient_id);
+                  if (removeButton) {
+                    removeButton.click();
+                  }
+                  altAddButton.click();
+                }
+                altReplaceButton.textContent = "Replace";
+                altReplaceButton.addEventListener("click", altReplaceFunction);
+                altReplaceCell.appendChild(altReplaceButton);
+              }
+            } else {
+              altTable.innerHTML = "None";
+            }
```

Comment:
Extracting and adopting the `TableBuilder` into a separate class could have populated the same table in fewer lines.